export_libraries.php

```diff
@@ -0,0 +1,172 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Export all libraries script.
+ *
+ * @package   mod_hvp
+ * @author    Rossco Hellmans <rosscohellmans@catalyst-au.net>
+ * @copyright Catalyst IT, 2021
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once('../../config.php');
+require_once($CFG->libdir . '/adminlib.php');
+require_once('locallib.php');
+
+// No guest autologin.
+require_login(0, false);
+
+admin_externalpage_setup('h5plibraries');
+
+$COURSE = $SITE;
+
+$interface = mod_hvp\framework::instance('interface');
+$core = mod_hvp\framework::instance('core');
+$exporter = new H5PExport($interface, $core);
+
+$libraries = $core->h5pF->loadLibraries();
+$tmppath = $exporter->h5pC->fs->getTmpPath();
+mkdir($tmppath, 0777, true);
```

Comment:
Should we wrap this inside a make_request dir or something so that its always cleaned up irregardless of H5P?

Reply:
Good idea, I've switched all the H5P tmp path stuff over to use make_request_directory.